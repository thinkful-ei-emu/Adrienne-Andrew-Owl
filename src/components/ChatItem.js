import React from 'react';

function ChatItem(props) {
  const participant = props.participant;
  const chatEvent = props.chatEvent;

  if (!participant || !chatEvent) {
    return (
      <div></div>
    )
  }
  const time = chatEvent.timestamp && new Date(chatEvent.timestamp);
  const startTime = chatEvent.time && new Date(chatEvent.time);
  
  // check chatEvent type and display accordingly
  const message = (type) => {
    const name = participant.name
    switch(type) {
      case 'message':
        return chatEvent.message;
      case 'thumbs-up':
        return `${name} gave a thumbs up b`;
      case 'thumbs-down':
        return `${name} gave a thumbs down p`;
      case 'raise-hand':
        return `${name} raised their hand o/`;
      case 'clap':
        return `${name} clapped /\\`;
      case 'join':
        return `${name} joined`;
      case 'leave':
        return `${name} left`;
      case 'join-stage':
        return `${name} joined the stage`;
      case 'leave-stage':
        return `${name} left the stage`;
      default:
        return '';
    }
  };

  return <>
    {startTime && <p>{startTime.toLocaleTimeString()}</p>}
    <div classNameName='chatItem'>
      <img className='chatAvatar' src={participant.avatar} alt='avatar here'></img>
      <p className='participant'>
        {participant.name}&nbsp;
        {time && 
        <span className='time'>{time.toLocaleTimeString()}</span>
        }
      </p>
      <p className={chatEvent.type === 'message' ? 'chatMessage': 'chatReaction'}>{message(chatEvent.type)}</p>
    </div>
   </>
}

export default ChatItem;