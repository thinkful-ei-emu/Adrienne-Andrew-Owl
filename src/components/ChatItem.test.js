import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatItem from './ChatItem';
import participants from '../data/participantData';
import chatEvents from '../data/chatEventsData';

// smoke test; failing time
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ChatItem key={0} chatEvents={{}} participant={{}} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// screenshot test;
it('renders this UI as expected', () => {
  const appTest = renderer.create(<ChatItem key={0} chatEvent={chatEvents[0]} participant={participants[0]}/>).toJSON();
  // Check whether it matches the previous snapshot
  // Stored in __snapshots__/App.test.js.snap
  expect(appTest).toMatchSnapshot(); 
});