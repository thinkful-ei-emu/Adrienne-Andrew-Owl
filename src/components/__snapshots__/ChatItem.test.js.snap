// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders this UI as expected 1`] = `
Array [
  <p>
    6:50:46 AM
  </p>,
  <div
    classNameName="chatItem"
  >
    <img
      alt="avatar here"
      className="chatAvatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <p
      className="participant"
    >
      Koren Templeton
      Â 
      <span
        className="time"
      >
        6:48:04 AM
      </span>
    </p>
    <p
      className="chatMessage"
    >
      Hello world
    </p>
  </div>,
]
`;
